src/canvas-view/canvas/canvas-component.js

```diff
@@ -0,0 +1,406 @@
+          }
+          // Don't let the chart handle the mouse down.
+          evt.stopPropagation();
+          evt.preventDefault();
+        };
+
+        //
+        // Handle connectorMouseOver on an connector.
+        //
+        ctrl.connectorMouseOver = function (evt, node, connector) {
+          if (!ctrl.readOnly) {
+            ctrl.mouseOverConnector = connector;
+          }
+        };
+
+        //
+        // Handle connectorMouseLeave on an connector.
+        //
+        ctrl.connectorMouseLeave = function () {
+          ctrl.mouseOverConnector = null;
+        };
+
+        //
+        // Handle mousedown on an input connector.
+        //
+        ctrl.connectorMouseDown = function (evt, node) {
+          if (ctrl.chart.inConnectingMode && node !== ctrl.connectingModeSourceNode) {
+            ctrl.chart.createNewConnection(ctrl.connectingModeOutputConnector, ctrl.mouseOverConnector);
+            ctrl.stopConnectingMode();
+          }
+        };
+
+        //
+        // zoom.
+        //
+        $scope.$on('zoomIn', function () {
+          ctrl.chart.zoom.in();
+        });
+
+        $scope.$on('zoomOut', function () {
+          ctrl.chart.zoom.out();
+        });
+
+        $scope.maxZoom = function () {
+          return (ctrl.chart.chartViewModel && ctrl.chart.chartViewModel.zoom) ? ctrl.chart.chartViewModel.zoom.isMax() : false;
+        };
+        $scope.minZoom = function () {
+          return (ctrl.chart.chartViewModel && ctrl.chart.chartViewModel.zoom) ? ctrl.chart.chartViewModel.zoom.isMin() : false;
+        };
+
+        ctrl.zoomLevel = function () {
+          return ctrl.chart.zoom.getLevel();
+        };
+
+        ctrl.$onInit = function () {
+          var backspaceKeyCode = 8;
+          var deleteKeyCode = 46;
+          var ctrlKeyCode = 17;
+          var aKeyCode = 65;
+          var escKeyCode = 27;
+
+          $document.find('body').keydown(function (evt) {
+            if (evt.ctrlKey === true) {
+              evt.stopPropagation();
+              evt.preventDefault();
+            }
+
+            if (evt.keyCode === aKeyCode && evt.ctrlKey === true) {
+              //
+              // Ctrl + A
+              //
+              ctrl.selectAll();
+              $scope.$digest();
+              evt.stopPropagation();
+              evt.preventDefault();
+            }
+          });
+
+          $document.find('body').keydown(function (evt) {
+            if (evt.keyCode === deleteKeyCode || evt.keyCode === backspaceKeyCode) {
+              ctrl.deleteSelected();
+              $scope.$digest();
+            }
+
+            if (evt.keyCode === escKeyCode) {
+              ctrl.deselectAll();
+              $scope.$digest();
+            }
+
+            if (evt.keyCode === ctrlKeyCode) {
+              evt.ctrlKey = false;
+              evt.stopPropagation();
+              evt.preventDefault();
+            }
+          });
```

Comment:
Just a note you don't need to change it. 

This could have also been flipped to a switch. However it probably wouldn't save us much...

```js
switch(evt.keyCode) {
  case deleteKeyCode:
  case backspaceKeyCode:
    ctrl.deleteSelected();
    $scope.$digest();
    break;

  case escKeyCode:
    ctrl.deselectAll();
    $scope.$digest();
    break;

  case ctrlKeyCode:
    evt.ctrlKey = false;
    evt.stopPropagation();
    evt.preventDefault();
    break;
}
```

Reply:
I'm fine with changing it to a switch. Looks nice and clean.